### ISSUE TITLE
Not compatible with Spectrum since generatetoaddress is not implemented

### ISSUE THREAD

**Issue by moneymanolis** [stepansnigirev/specterext-faucet#1]
<img width="929" alt="grafik" src="https://user-images.githubusercontent.com/70536101/204361889-68cd66ea-50dd-4140-9541-387e3b403361.png">

**Comment by stepansnigirev** [comment]
True, there is no such call in Electrum API, so there is no way to generate a new block via Spectrum.
I think this is unfixable.

**Comment by moneymanolis** [comment]
OK, I see, then we will just have to check whether Spectrum is used and make a warning, similar to the exfund extension that can't be used without a wallet.

**Comment by k9ert** [comment]
Better just make a reasonable error-message. It's not worth to increase complexity for a feature which is only existing on regtest.

**Comment by PaarthAgarwal** [comment]
> Better just make a reasonable error-message. It's not worth to increase complexity for a feature which is only existing on regtest.

I'd like to take on this issue but will need a little help as I'm new to this codebase. What should be the reasonable error message?

**Comment by k9ert** [comment]
Great! I'd suggest to:
* Choose one of the lifecycle callback methods ([here](https://docs.specter.solutions/desktop/extensions/callbacks/) ) e.g. `after_serverpy_init_app` 
* implement it so that it checks whether `generatetoaddress` is available and store that information somewhere, probably a property of the service object
* Check in the controller whether we can generate coins at all via that property and if not, redirect to a page which is explaining that.

If you have any further question, the developer community chat is a great place to ask.
